### PR TITLE
feature: handle too many players

### PR DESCRIPTION
**Pull request by ct222ct**

### CODE REVIEW

**Comment by lechld** [comment]
Screenshot 2023-05-11 at 20 36 05

As I shared with you, please do basic formatting. That skips a lot of unnecessary reviewing as we don't see changes that "are no changes"

File: app/src/main/java/at/aau/edu/wizards/ui/lobby/LobbyViewModel.kt [hunk 1]
```diff
@@ -29,18 +34,26 @@ class LobbyViewModel(
     fun stopAdvertising() {
         server.stopBroadcasting()
     }
-
+    var numplayer = 1
+    var check_too_many_player = false
```
Comment (lechld) [inline app/src/main/java/at/aau/edu/wizards/ui/lobby/LobbyViewModel.kt:38]: So many warnings here. @Kirschzelle always points them out. Main issue is that they are warnings in sonar. And btw they have a reason to be pointed out by the IDE

Comment (lechld) [reply app/src/main/java/at/aau/edu/wizards/ui/lobby/LobbyViewModel.kt:38]: Screenshot 2023-05-11 at 21 03 47

Welle drunter --- nicht zum Spaß da

File: app/src/test/java/at/aau/edu/wizards/discover_testing/discovertesting.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import at.aau.edu.wizards.api.impl.ClientImpl
+import at.aau.edu.wizards.api.impl.MessageDelegate
+import at.aau.edu.wizards.ui.discover.DiscoverViewModel
+import com.google.android.gms.nearby.connection.ConnectionsClient
+import org.junit.jupiter.api.Test
+import org.mockito.kotlin.spy
+import org.mockito.kotlin.times
+import org.mockito.kotlin.verify
+
+class discovertesting {
```
[the remark this answers is not in the document]
Comment (ct222ct) [reply app/src/test/java/at/aau/edu/wizards/discover_testing/discovertesting.kt:12]: I have deleted the file

File: app/src/main/java/at/aau/edu/wizards/ui/lobby/LobbyViewModel.kt (hunk 1 above)
Comment (lechld) [reply app/src/main/java/at/aau/edu/wizards/ui/lobby/LobbyViewModel.kt]: please resolve conversations when you fixed them

File: app/src/main/java/at/aau/edu/wizards/ui/lobby/LobbyViewModel.kt [hunk 3]
```diff
@@ -30,17 +30,26 @@ class LobbyViewModel(
         server.stopBroadcasting()
     }
 
+    var numplayer = 1
```
Comment (lechld) [inline app/src/main/java/at/aau/edu/wizards/ui/lobby/LobbyViewModel.kt:33]: Just from naming convention `numPlayer` and `checkTooManyPlayer`, nothing crucial

File: app/src/main/java/at/aau/edu/wizards/ui/lobby/LobbyViewModel.kt [hunk 4]
```diff
@@ -30,17 +30,26 @@ class LobbyViewModel(
...
+    var checktoomanyplayer = false
```
Comment (lechld) [inline app/src/main/java/at/aau/edu/wizards/ui/lobby/LobbyViewModel.kt:34]:
```
var propertyName: String = "abc"
    private set // the setter is private and has the default implementation
```

You should use something like that for both properties. It's basically equivalent to following java code

```
private boolean checkTooManyPlayer = false;

public boolean getCheckTooManyPlayer() {
    return checkTooManyPlayer;
}
```

**Review by lechld — COMMENTED** [review]
Please add unit test for added logic inside ViewModel

File: app/src/main/java/at/aau/edu/wizards/ui/lobby/LobbyViewModel.kt (hunk 3 above)
Comment (ct222ct) [reply app/src/main/java/at/aau/edu/wizards/ui/lobby/LobbyViewModel.kt:33]: completed

File: app/src/main/java/at/aau/edu/wizards/ui/lobby/LobbyViewModel.kt (hunk 4 above)
Comment (ct222ct) [reply app/src/main/java/at/aau/edu/wizards/ui/lobby/LobbyViewModel.kt]: completed

File: app/src/main/java/at/aau/edu/wizards/ui/lobby/LobbyViewModel.kt [hunk 5]
```diff
@@ -30,17 +30,33 @@ class LobbyViewModel(
         server.stopBroadcasting()
     }
 
+    var numPlayer: Int = 1
+        private set
```
Comment (lechld) [inline app/src/main/java/at/aau/edu/wizards/ui/lobby/LobbyViewModel.kt:34]: here you used `private set`

File: app/src/main/java/at/aau/edu/wizards/ui/lobby/LobbyViewModel.kt [hunk 6]
```diff
@@ -30,17 +30,33 @@ class LobbyViewModel(
...
+
+    private var checkTooManyPlayer = false
+
+    fun getCheckTooManyPlayer(): Boolean {
```
Comment (lechld) [inline app/src/main/java/at/aau/edu/wizards/ui/lobby/LobbyViewModel.kt:38]: remove that function write `private set` in line 37. Exactly same behaviour, that's what I meant.

File: app/src/main/res/drawable/ic_launcher_background.xml [hunk 7]
```diff
@@ -1,74 +1,170 @@
 <?xml version="1.0" encoding="utf-8"?>
```
Comment (lechld) [inline app/src/main/res/drawable/ic_launcher_background.xml:1]: why do you change the launcher icon?  Have you used it somewhere?

Seems you found a way to format whole project. And that is nice, but should be done in a separate pull request. Because what now happens - I need to check every line in every file you changed

It still has a reason why I shared that marco

File: app/src/test/java/at/aau/edu/wizards/TooManyPlayerUnitTest.kt [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+import at.aau.edu.wizards.ui.lobby.LobbyItem
+import at.aau.edu.wizards.ui.lobby.LobbyItemFactory
+import at.aau.edu.wizards.ui.lobby.LobbyViewModel
+import junit.framework.TestCase.assertFalse
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+import org.mockito.kotlin.mock
+
+class TooManyPlayerUnitTest {
```
Comment (lechld) [inline app/src/test/java/at/aau/edu/wizards/TooManyPlayerUnitTest.kt:14]: `LobbyViewModelTest`. You write tests for classes present, alway just suffix them with `ClassName...Test`

File: app/src/test/java/at/aau/edu/wizards/TooManyPlayerUnitTest.kt [hunk 9]
```diff
@@ -0,0 +1,66 @@
+package at.aau.edu.wizards
```
Comment (lechld) [inline app/src/test/java/at/aau/edu/wizards/TooManyPlayerUnitTest.kt:1]: Please move the package to the class you want to test. Checkout some of the `api.impl` tests otherwise please

File: app/src/test/java/at/aau/edu/wizards/gameModel/GameModelPlayerUnitTest.kt [hunk 10]
```diff
@@ -93,4 +94,6 @@ class GameModelPlayerUnitTest {
         assertEquals(-100, player.scores[4])
         assertEquals(100, player.scores[5])
     }
+
+
```
Comment (lechld) [inline app/src/test/java/at/aau/edu/wizards/gameModel/GameModelPlayerUnitTest.kt:98]: One more file to check, every change... But why adding two empty lines here? One line would be correct, I see that.

File: app/src/test/java/at/aau/edu/wizards/TooManyPlayerUnitTest.kt [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+        viewModel.clicked(LobbyItem.AddCpu)
+        viewModel.clicked(LobbyItem.AddCpu)
+        viewModel.clicked(LobbyItem.Requested(requestedConnection))
+        viewModel.clicked(LobbyItem.AddCpu)
+        viewModel.clicked(LobbyItem.AddCpu)
+        viewModel.clicked(LobbyItem.AddCpu)
+
+        assertTrue(viewModel.getCheckTooManyPlayer())
+    }
+
```
Comment (lechld) [inline app/src/test/java/at/aau/edu/wizards/TooManyPlayerUnitTest.kt:65]: You missed the case where it's neither `AddCpu` or `Requested`. That's why we have reports showing it to us, please add that too. It's not a perfect test case right now, but good work 👍. Please add the missing piece and we achieve the goal of "having code coverage". 

Screenshot 2023-05-12 at 21 30 48

File: app/src/main/java/at/aau/edu/wizards/ui/lobby/LobbyViewModel.kt (hunk 6 above)
Comment (ct222ct) [reply app/src/main/java/at/aau/edu/wizards/ui/lobby/LobbyViewModel.kt:38]: completed

File: app/src/test/java/at/aau/edu/wizards/TooManyPlayerUnitTest.kt (hunk 8 above)
Comment (ct222ct) [reply app/src/test/java/at/aau/edu/wizards/TooManyPlayerUnitTest.kt:14]: completed

File: app/src/test/java/at/aau/edu/wizards/TooManyPlayerUnitTest.kt (hunk 11 above)
Comment (ct222ct) [reply app/src/test/java/at/aau/edu/wizards/TooManyPlayerUnitTest.kt]: everything is now tested

**Review by lechld — APPROVED** [review]
